src/Alert.php

```diff
@@ -121,15 +170,128 @@ public function options(array $value): self
     }
 
     /**
-     * Renders the alert body and the close button (if any).
+     * Enable/Disable encode body
      *
-     * @throws JsonException
+     * @param bool $encode
      *
-     * @return string the rendering result
+     * @return self
+     */
+    public function encode(bool $encode = true): self
+    {
+        $new = clone $this;
+        $new->encode = $encode;
+
+        return $new;
+    }
+
+    /**
+     * Enable/Disable dissmiss animation
+     *
+     * @param bool $fade
+     *
+     * @return self
+     */
+    public function fade(bool $fade = true): self
```

Comment:
Instead have a boolean variable why not add the class directly to options example:

```php
public function fade(): self
{
    $new = clone $this;
    Html::addCssClass($new->options, 'fade show');
    return $new;
}

This i can do for all widgets they are simple classes that don't need a variable or constant.

Reply:
With this it can't be removed 
```
$alert = Alert::widget()->fade();

echo $alert->success();
....
echo $alert->danger()->fade(false); //?
```

Reply:
Because if I don't call the `fade()` function, it should not set anything by default.

Reply:
> Because if I don't call the `fade()` function, it should not set anything by default.

Maybe `withFade`?

Reply:
Well I think in this case `fade(false)` is better, it is more readable,